runtime/lib/mirrors.cc

```diff
@@ -835,12 +835,49 @@ DEFINE_NATIVE_ENTRY(Mirrors_makeLocalClassMirror, 1) {
                            Object::null_instance());
 }
 
-
 DEFINE_NATIVE_ENTRY(Mirrors_makeLocalTypeMirror, 1) {
   GET_NON_NULL_NATIVE_ARGUMENT(AbstractType, type, arguments->NativeArgAt(0));
   return CreateTypeMirror(type);
 }
 
+DEFINE_NATIVE_ENTRY(Mirrors_makeLocalTypeMirrorWithTypeArguments, 2) {
+  GET_NON_NULL_NATIVE_ARGUMENT(AbstractType, type, arguments->NativeArgAt(0));
+  GET_NON_NULL_NATIVE_ARGUMENT(Array, args, arguments->NativeArgAt(1));
+
+  const Class& clz = Class::Handle(type.type_class());
+  if (!clz.IsGeneric()) {
+    const Array& errorArgs = Array::Handle(Array::New(3));
+    errorArgs.SetAt(0, type);
+    errorArgs.SetAt(1, String::Handle(String::New("key")));
+    errorArgs.SetAt(2, String::Handle(String::New("Type must be a generic class")));
```

Comment:
Hmmm.
Grepping through our sources, I can unfortunately see both. Even worse, the errors themselves sometimes add a dot to the message and sometimes they don't.
ArgumentError doesn't add a dot, so it should be safe to add one.
There is really no consistency right now. Do as you please.
We will (hopefully) eventually go through them and clean them up.